pytorch_lightning/metrics/sklearn.py

```diff
@@ -0,0 +1,715 @@
+from typing import Any, Optional, Union, Sequence
+
+import numpy as np
+import torch
+
+from pytorch_lightning import _logger as lightning_logger
+from pytorch_lightning.metrics.metric import NumpyMetric
+
+__all__ = [
+    'SklearnMetric',
+    'Accuracy',
+    'AveragePrecision',
+    'AUC',
+    'ConfusionMatrix',
+    'F1',
+    'FBeta',
+    'Precision',
+    'Recall',
+    'PrecisionRecallCurve',
+    'ROC',
+    'AUROC'
+]
+
+
+class SklearnMetric(NumpyMetric):
+    """
+    Bridge between PyTorch Lightning and scikit-learn metrics
+
+        Warning:
+            Every metric call will cause a GPU synchronization, which may slow down your code
+
+        Note:
+            The order of targets and predictions may be different from the order typically used in PyTorch
+    """
+    def __init__(self, metric_name: str,
+                 reduce_group: Any = torch.distributed.group.WORLD,
+                 reduce_op: Any = torch.distributed.ReduceOp.SUM, **kwargs):
+        """
+        Args:
+            metric_name: the metric name to import and compute from scikit-learn.metrics
+            reduce_group: the process group for DDP reduces (only needed for DDP training).
+                Defaults to all processes (world)
+            reduce_op: the operation to perform during reduction within DDP (only needed for DDP training).
+                Defaults to sum.
+            **kwargs: additonal keyword arguments (will be forwarded to metric call)
+        """
+        super().__init__(name=metric_name, reduce_group=reduce_group,
+                         reduce_op=reduce_op)
+
+        self.metric_kwargs = kwargs
+
+        lightning_logger.debug(
+            'Every metric call will cause a GPU synchronization, which may slow down your code')
```

Comment:
Should this be maybe be changed to `lightning_logger.warning`? 
Also I would suggest that we made the text metric specific:
```suggestion
            f'Metric {self.__class__.__name__} is using Sklearn as backend, meaning that'
            ' every metric call will cause a GPU synchronization, which may slow down your code'
            
```